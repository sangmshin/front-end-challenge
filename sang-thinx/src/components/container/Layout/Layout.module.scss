@import 'variables';

.root {
  // padding: 0 60px;
}

@media (min-width: $screen-md) {
  .root {
    // padding: 0 90px;
  }
}