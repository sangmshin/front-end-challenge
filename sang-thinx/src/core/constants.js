export const COLOR_HEXCODE_MAP = {
  beige: '#f2d9a5',
  black: '#000',
}