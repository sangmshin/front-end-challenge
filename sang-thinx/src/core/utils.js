const SMALL = 768; /* Small screen / tablet */
const MEDIUM = 1024; /* Medium screen / desktop */
const LARGE = 1200; /* Large screen / wide desktop */
let BREAKPOINT;

/**
 * Finds the current matched media query breakpoint
 *
 * @returns {String} matched media query breakpoint [large, medium, small, xsmall]
 */
export function getCurrentBreakPoint() {
  if (!BREAKPOINT) {
    BREAKPOINT = {
      large: window.matchMedia(`(min-width: ${LARGE}px)`),
      medium: window.matchMedia(`(min-width: ${MEDIUM}px)`),
      small: window.matchMedia(`(min-width: ${SMALL}px)`),
      xsmall: window.matchMedia(`(max-width: ${SMALL - 1}px)`)
    };
  }

  return Object.keys(BREAKPOINT).find(breakpoint => BREAKPOINT[breakpoint].matches);
}

/**
 * Determines if the data is array and has at least one or more items
 * @return {boolean}
 */
export const isArrayWithLength = arr => Array.isArray(arr) && arr.length > 0;

/**
 * A function with no operation. (for testing)
 */
export const noop = () => null;
