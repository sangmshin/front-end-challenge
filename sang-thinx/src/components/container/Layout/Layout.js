import React, { useEffect } from 'react';
import { func, string } from 'prop-types';
import { connect } from 'react-redux';
import debounce from 'debounce';
import { getPageData } from 'store/page-data/actions';
import { setLastBreakPoint } from 'store/ui/actions';
import { getLastBreakPoint } from 'store/ui/selectors';
import { getCurrentBreakPoint } from 'core/utils';
import styles from './Layout.module.scss';

const Layout = ({
  children,
  getPageData,
  lastBreakPoint,
  setLastBreakPoint,
}) => {
  const handleResize = () => {
    const newBreakpoint = getCurrentBreakPoint();
    if (newBreakpoint !== lastBreakPoint) {
      setLastBreakPoint(newBreakpoint);
    }
  };
  const debouncedHandleResize = debounce(handleResize, 250);

  useEffect(() => {
    window.addEventListener('resize', debouncedHandleResize);
    handleResize();
    getPageData();
    console.log('mounted');

    return () => window.removeEventListener('resize', debouncedHandleResize);
  }, [])

  return (
    <div className={styles.root}>
      {children}
    </div>
  )
}

Layout.propTypes = {
  getPageData: func.isRequired,
  lastBreakPoint: string,
  setLastBreakPoint: func.isRequired
}

Layout.defaultProps = {
  lastBreakPoint: ''
}

const mapStateToProps = state => ({
  lastBreakPoint: getLastBreakPoint(state)
})

const mapDispatchToProps = {
  getPageData,
  setLastBreakPoint
};

export default connect(mapStateToProps, mapDispatchToProps)(Layout);