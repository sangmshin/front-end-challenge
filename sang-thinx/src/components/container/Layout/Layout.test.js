import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Layout } from './Layout';
import { noop } from 'core/utils';

const props = {
  getPageData: noop,
  setLastBreakPoint: noop,
}

describe('Component :: Layout', () => {
  const wrapper = shallow(
    <Layout {...props} >
      <div className="child" />
    </Layout>
  );

  describe('#render', () => {
    it('should match snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
