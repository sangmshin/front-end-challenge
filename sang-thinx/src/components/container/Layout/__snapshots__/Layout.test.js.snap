// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component :: Layout #render should match snapshot 1`] = `
<div>
  <Connect(Zoom) />
  <div
    className="child"
  />
</div>
`;
